lib/spack/spack/build_systems/cmake.py

```diff
@@ -32,6 +32,50 @@
 from spack.package import PackageBase, run_after
 
 
+def processCMakeCacheFile(cache_file):
+    def determineLineInterest(line):
+        return not line.startswith("//") \
+            and not line.startswith("#") \
+            and len(line) > 0 \
+            and not ":INTERNAL" in line \
+            and not ":PATH" in line \
+            and not ":FILEPATH" in line \
+            and not ":STATIC" in line \
+            and not "CMAKE_" in line \
+            and line.count("/") < 4  # *really* don't want to capture paths
+
+    def dictionaryFromLines(lines):
```

Comment:
I know there are places where inner functions are anathema, are they kosher in Spack? I see these possibly being changed or used multiple times